lib/buildreq.c

```diff
@@ -175,7 +175,7 @@ int rc_aaa_ctx_server(rc_handle * rh, RC_AAA_CTX ** ctx, SERVER * aaaserver,
 
 		result = rc_send_server_ctx(rh, ctx, &data, msg, type);
 
-		if ((result == OK_RC) || (result == CHALLENGE_RC)) {
```

Comment:
You seem to be right about the av-pair list. I verified that code typically frees that recvd variable unconditionally. We may need some documentation improvement there.